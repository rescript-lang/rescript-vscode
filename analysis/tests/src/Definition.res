let xx = 10

let y = xx
//      ^def

module Inner = {
  type tInner = int
  let vInner = 34
}

type typeInner = Inner.tInner
//                     ^def

// open Belt
let m1 = List.map
//            ^hov

open ShadowedBelt
let m2 = List.map
//            ^hov

let uncurried = (. x, y) => x + y

uncurried(. 3, 12)->ignore
// ^hov

uncurried(. 3, 12)->ignore
// ^def