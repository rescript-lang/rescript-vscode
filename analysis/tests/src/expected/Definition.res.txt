Definition src/Definition.res 2:8
{"uri": "Definition.res", "range": {"start": {"line": 0, "character": 4}, "end": {"line": 0, "character": 6}}}

Definition src/Definition.res 10:23
{"uri": "Definition.res", "range": {"start": {"line": 6, "character": 7}, "end": {"line": 6, "character": 13}}}

Hover src/Definition.res 14:14
{"contents": {"kind": "markdown", "value": "```rescript\n('a => 'b, list<'a>) => list<'b>\n```\n\n [List.map f [a1; ...; an]] applies function [f] to [a1, ..., an],\n   and builds the list [[f a1; ...; f an]]\n   with the results returned by [f].  Not tail-recursive. "}}

Hover src/Definition.res 18:14
{"contents": {"kind": "markdown", "value": "```rescript\n(list<'a>, 'a => 'b) => list<'b>\n```"}}

Hover src/Definition.res 23:3
{"contents": {"kind": "markdown", "value": "```rescript\n(int, int) => int\n```"}}

Definition src/Definition.res 26:3
{"uri": "Definition.res", "range": {"start": {"line": 21, "character": 4}, "end": {"line": 21, "character": 13}}}

